Figure out how to get it to work with jsdom

Would be much more lightweight using [jsdom](https://github.com/tmpvar/jsdom) than the current PhantomJS setup.

This is how far I got. Currently only returns `0`, which means the JS on the site is never run for some reason.

``` js
'use strict';
const jsdom = require('jsdom');

jsdom.env({
    url: 'http://fast.com/',
    scripts: ['http://code.jquery.com/jquery.js'],
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/50.0.2661.102 Safari/537.36',
    strictSSL: false,
    features: {
        FetchExternalResources: ['script'],
        ProcessExternalResources: ['script']
    },
    done: (err, window) => {
        if (err) {
            throw err;
        }

        setTimeout(() => {
            console.log('Speed:', window.$('#speed-value').text().trim(), window.$('#speed-units').text().trim());
        }, 5000);
    }
});
```
## 

Help appreciated 😀

Don't know if it acceptable practice to post this comment here, but since this is where I landed looking for info related to fast-cli and PahntomJS...

`npm` installation fails on Raspberry PI (3) due to no native ARM PhantomJS. I attempted to compile based on the instructions at the PhantomJS site and it failed in an ugly fashion:

```sh
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
Makefile.WebCore.Target:47888: recipe for target '.obj/page/DOMWindow.o' failed
make[2]: *** [.obj/page/DOMWindow.o] Error 4
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/pippipi/phantomjs/src/qt/qtwebkit/Source/WebCore'
Makefile.WebCore:68: recipe for target 'sub-Target-pri-make_first-ordered' failed
make[1]: *** [sub-Target-pri-make_first-ordered] Error 2
make[1]: Leaving directory '/home/pippipi/phantomjs/src/qt/qtwebkit/Source/WebCore'
Makefile:177: recipe for target 'sub-Source-WebCore-WebCore-pro-make_first-ordered' failed
make: *** [sub-Source-WebCore-WebCore-pro-make_first-ordered] Error 2

ERROR: Failed to build PhantomJS! Building Qt WebKit failed.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,4 +1,4 @@
 language: node_js
 node_js:
+  - '8'
   - '6'
-  - '4'

--- a/api.js
+++ b/api.js
@@ -1,19 +1,16 @@
 'use strict';
 /* eslint-env browser */
-const driver = require('promise-phantom');
-const phantomjs = require('phantomjs-prebuilt');
+const puppeteer = require('puppeteer');
 const Observable = require('zen-observable');
 
-function init(page, observer, prevSpeed) {
-	// TODO: Doesn't work with arrow function. open issue on `promise-phantom`
-	page.evaluate(function () { // eslint-disable-line prefer-arrow-callback
+function init(page, browser, observer, prevSpeed) {
+	page.evaluate(() => {
 		const $ = document.querySelector.bind(document);
 
 		return {
 			speed: Number($('#speed-value').textContent),
 			unit: $('#speed-units').textContent.trim(),
-			// Somehow it didn't work with `Boolean($('#speed-value.succeeded'))`
-			isDone: document.querySelectorAll('.succeeded').length > 0
+			isDone: Boolean($('#speed-value.succeeded'))
 		};
 	})
 	.then(result => {
@@ -22,20 +19,19 @@ function init(page, observer, prevSpeed) {
 		}
 
 		if (result.isDone) {
-			page.close();
+			browser.close();
 			observer.complete();
 		} else {
-			setTimeout(init, 100, page, observer, result.speed);
+			setTimeout(init, 100, page, browser, observer, result.speed);
 		}
 	})
 	.catch(err => observer.error(err));
 }
 
 module.exports = () => new Observable(observer => {
-	driver.create({path: phantomjs.path})
-		.then(phantom => phantom.createPage())
-		.then(page => page.open('http://fast.com').then(() => {
-			init(page, observer);
-		}))
+	puppeteer.launch()
+		.then(browser => Promise.all([browser, browser.newPage()]))
+		.then(([browser, page]) => Promise.all([browser, page, page.goto('http://fast.com')]))
+		.then(([browser, page]) => init(page, browser, observer))
 		.catch(err => observer.error(err));
 });

--- a/cli.js
+++ b/cli.js
@@ -49,22 +49,9 @@ if (process.stdout.isTTY) {
 	}, 50);
 }
 
-let timeout;
-
 api()
 	.forEach(result => {
 		data = result;
-		// Exit after the speed has been the same for 3 sec
-		// needed as sometimes `isDone` doesn't work for some reason
-		clearTimeout(timeout);
-		timeout = setTimeout(() => {
-			data.isDone = true;
-			exit();
-		}, 5000);
-
-		if (data.isDone) {
-			exit();
-		}
 	})
 	.then(() => exit())
 	.catch(err => {

--- a/package.json
+++ b/package.json
@@ -13,7 +13,7 @@
     "fast": "cli.js"
   },
   "engines": {
-    "node": ">=4"
+    "node": ">=6.4.0"
   },
   "scripts": {
     "test": "xo && ava"
@@ -45,8 +45,7 @@
     "log-update": "^1.0.0",
     "meow": "^3.3.0",
     "ora": "^1.2.0",
-    "phantomjs-prebuilt": "^2.1.7",
-    "promise-phantom": "^3.1.1",
+    "puppeteer": "^0.11.0",
     "zen-observable": "^0.5.1"
   },
   "devDependencies": {

--- a/readme.md
+++ b/readme.md
@@ -7,7 +7,7 @@
 
 ## Install
 
-Ensure you have [Node.js](https://nodejs.org) version 4+ installed. Then run the following:
+Ensure you have [Node.js](https://nodejs.org) version 6.4+ installed. Then run the following:
 
 ```
 $ npm install --global fast-cli

